g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/TranslationProductComponentAction.java

```diff
@@ -201,8 +201,14 @@ private String getMatchedVersion(final String productName, final String version)
 				}
 				for(String s : versionList) {
 					String f = filterVersion(s, version);
-					String ss = StringUtils.isEmpty(matchedVersion) ? s : filterVersion(s, matchedVersion);
-					if((compare(f, version) == -1 || (compare(f, version) == 0 && s.length() < version.length())) && compare(ss, matchedVersion) == 1) {
+					String ss = s;
+					String m = matchedVersion;
+					if(!StringUtils.isEmpty(matchedVersion)) {
+						if(ss.split("\\.").length > matchedVersion.split("\\.").length) {
+							m = filterVersion(matchedVersion, ss);
+						}
+					}
+					if((compare(f, version) == -1 || (compare(f, version) == 0 && s.length() < version.length())) && compare(ss, m) == 1) {
```

Comment:
It's a little difficult to understand this snippet's logic since 6 variables(version, s, f, ss, m, matchedVersion ) are referred to and their values are always changed, is it possible to refactor to make it clear or add comment to explain what each variable represents? especially the meaning of f, ss, m.

It's much better add comment for line 211 since the condition is a little complex.